### PR TITLE
Remove KoreBuild dependencies in build steps

### PR DESCRIPTION
**Pull request by JunTaoLuo**
This will fix our RuntimeStore_Installer builds. Also cleaning up by removing a few targets/build steps that are no longer needed.

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
Hmm seems like I broke something in NuGetPackageVerifier. Investigating.

File: build/repo.targets [hunk 1]
```diff
@@ -876,30 +852,21 @@
       Command="docker build --build-arg USER_ID=%24(id -u) -t docker-image-$(InstallerPlatform) $(InstallerPlatform)"
       WorkingDirectory="$(DockerDir)" />
 
-    <!-- Parse CLI version -->
-    <ReadLinesFromFile File="$(BuildDirectory)cli.version">
-      <Output TaskParameter="Lines" PropertyName="CLIVersionNumber" />
-    </ReadLinesFromFile>
+    <!-- Delete korebuild-lock.txt to trigger re-installation of dotnet -->
+    <Delete Files="$(RepositoryRoot)korebuild-lock.txt" />
```
Comment (natemcmaster) [inline build/repo.targets]: why? This is potential problematic. 😬 The lockfile should work without changes, even within the docker container.

**Comment by natemcmaster** [comment]
The NGPV bug is my fault. I rearranged the files in the buildtools repo -> namespace change -> embedded resource name changed.

File: build/repo.targets (hunk 1 above)
Comment (JunTaoLuo) [inline build/repo.targets]: Ah yes, I just wanted to retrigger installation. I think I saw an option to force installation though.

**Comment by natemcmaster** [comment]
https://github.com/aspnet/BuildTools/commit/ed5b9e5725186ee3f1a92177fb1fd79f5cf4a841

File: build/repo.targets (hunk 1 above)
Comment (JunTaoLuo) [inline build/repo.targets]: Let me try `--update`

Comment (JunTaoLuo) [inline build/repo.targets]: Hmm Seems like it's not needed actually.

**Pull request merged**